main/lsp/ErrorReporter.cc

```diff
@@ -155,7 +155,8 @@ void ErrorReporter::pushDiagnostics(uint32_t epoch, core::FileRef file, const ve
             tags.push_back(DiagnosticTag::Unnecessary);
             diagnostic->tags = move(tags);
         }
-        diagnostic->severity = DiagnosticSeverity::Error;
+
+        diagnostic->severity = toDiagnosticSeverity(error->severity);
```

Comment:
IMO, this is overabstracted.

Sorbet takes a somewhat strong philosophical stance that "warnings don't exist—there's errors and there's nothing"

in the case of this feature, while we're hijacking LSP's diagnostic reporting for this, which does have levels of errors, fundamentally the feature we're really trying to build (if it there were an API for that in the LSP spec) is to have some sort of view or layer on top of source code.

All of which is to say: I think that the API we program against, the core::Error and ErrorClass APIs, should not have a notion of severity.

Rather, at the point where we translate these errors to LSP diagnostics right here, we should ask, `error->what == <the special untyped code error class>`, and if so, set the severity to something else.

In particular, by not having a way to set a severity for an Error, people can't make (bad) arguments in the future like, "Why don't we just build <broken type system feature that should not exist at all>, but make it only a warning?"

(Also, I think we should call that severity Information, not Warning, which again drives home the point that it's not like there are some "optional errors.")

Reply:
ack, patched

Thanks for the detailed feedback and context!